ui/apps/platform/src/Containers/Collections/converter.ts

```diff
@@ -0,0 +1,177 @@
+                    type: 'ByName',
+                    field,
+                    rule: { operator: 'OR', values: [] },
+                };
+            }
+        }
+
+        const selector = collection.resourceSelector[entity];
+
+        switch (selector.type) {
+            case 'ByLabel': {
+                const firstValue = rule.values[0]?.value;
+
+                if (firstValue && firstValue.includes(LABEL_SEPARATOR)) {
+                    const key = firstValue.split(LABEL_SEPARATOR)[0] ?? '';
+                    selector.rules.push({
+                        operator: 'OR',
+                        key,
+                        // TODO Verify with BE whether or not this is a valid method to get the label values. Is
+                        //      it possible that multiple `=` symbols will appear in the data here?
+                        values: rule.values.map(({ value }) => value.split('=')[1] ?? ''),
+                    });
+                }
+                break;
+            }
+            case 'ByName':
+                selector.rule.values = rule.values.map(({ value }) => value);
+                break;
+            case 'All':
+                // Do nothing
+                break;
+            default:
+                ensureExhaustive(selector);
+        }
+    });
+
+    if (errors.length > 0) {
+        return new AggregateError(errors);
+    }
+
+    return collection;
+}
+
+/**
+ * This function takes the UI supported representation of a `Collection` and generates
+ * a server response in the generic `CollectionRequest` format supported by the back end.
+ */
+export function generateRequest(collection: Collection): CollectionRequest {
+    const rules: SelectorRule[] = [];
+
+    selectorEntityTypes.forEach((entityType) => {
+        const selector = collection.resourceSelector[entityType];
+
+        switch (selector.type) {
+            case 'ByName':
+                rules.push({
+                    fieldName: selector.field,
+                    operator: selector.rule.operator,
+                    values: selector.rule.values.map((value) => ({ value })),
+                });
+                break;
+            case 'ByLabel':
+                selector.rules.forEach(({ operator, key, values }) => {
+                    rules.push({
+                        fieldName: selector.field,
+                        operator,
+                        values: values.map((value) => ({
+                            value: `${key}${LABEL_SEPARATOR}${value}`,
+                        })),
+                    });
+                });
+                break;
+            case 'All':
+                // Append nothing
+                break;
+            default:
+                ensureExhaustive(selector);
+        }
+    });
+
+    return {
+        name: collection.name,
+        description: collection.description,
+        embeddedCollectionIds: collection.embeddedCollectionIds,
```

Comment:
Can you double check it the key is `embeddedCollectionIds` or `embeddedCollections`

https://github.com/stackrox/stackrox/blob/master/proto/storage/resource_collection.proto#L28

Reply:
I think the internal proto is different from what they will be sending/accepting at the service layer. I was basing the naming on the original BE design doc as well as following along with some of the PRs that get announced on the epic channel. It looks like for this type of request [they are going](https://github.com/stackrox/stackrox/pull/3600/files#diff-2abd114e43fc8626980140aecb4464f03014c29f5ca1fdf3c4ef88a7196caa7dR30) with the `ids` version.